### PR TITLE
[Spark] CI - Use Python packages compatible to Python 3.9 + Run CI when workflow files change

### PR DESCRIPTION
**Pull request by felipepessoto**
- [X] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description

Use Python packages compatible to both Python 3.8 and 3.9

https://setuptools.pypa.io/en/stable/history.html#v41-1-0
https://pypi.org/project/pandas/1.1.3/

### CODE REVIEW

**Comment by felipepessoto** [comment]
\+ @vkorukanti who made the last changes to this file.

**Comment by allisonport-db** [comment]
Can you push an empty commit just so we re-trigger all the CI jobs?

**Comment by felipepessoto** [comment]
@allisonport-db it is all green

File: .github/workflows/spark_test.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ jobs:
         with:
           PATTERNS: |
             **
+            .github/workflows/**
```
[the remark this answers is not in the document]
Comment (felipepessoto) [reply .github/workflows/spark_test.yaml:19]: Without this the tests are skipped for some reason:
https://github.com/delta-io/delta/pull/2729
image

[the remark this answers is not in the document]
Comment (felipepessoto) [reply .github/workflows/spark_test.yaml:19]: Maybe "steps.git-diff.outputs.diff" doesn't read folders starting with "."?

[the remark this answers is not in the document]
Comment (allisonport-db) [reply .github/workflows/spark_test.yaml:19]: Weird.. I thought I've seen it rerun with such changes in the past. But okay SGTM

**Review by allisonport-db — APPROVED** [review]

**Pull request merged**